### PR TITLE
refactor: add clipboard button to uikit [WEB-1122, WEB-355]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Add Clipboard Button to the UI Kit as it is a common enough of a pattern to reuse.

### Screenshots

Render Clipboard Button
Screenshot 2023-05-09 at 9 34 22 AM

Hover Over Button
Screenshot 2023-05-09 at 9 34 27 AM

Successful Copy
Screenshot 2023-05-09 at 9 34 35 AM

Failed to Copy with Reason
Screenshot 2023-05-09 at 9 35 10 AM

## Test Plan

Test areas that have been upgraded to the new clipboard pattern.
- [ ] Cluster > Master Logs - Copy to Clipboard (upper right corner button)
- [ ] Model Registry > Model > Model Version - Download (upper right corner button
- [ ] Model Registry > Model > Model Version - Use in Notebook (upper right corner button
- [ ] Experiments > Trial Detail > Logs - Copy to Clipboard (upper right corner button)
- [ ] Experiments > Trial Detail > Logs - Download (upper right corner button)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-1122
WEB-355

### CODE REVIEW

File: webui/react/src/components/kit/ClipboardButton.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return () => {
+      button?.removeEventListener('mouseenter', onMouseEnter);
+      button?.removeEventListener('mouseleave', onMouseLeave);
+    };
+  }, []);
+
+  return (
+    <Tooltip content={tooltipLabel} open={tooltipOpen} placement={tooltipPlacement}>
+      <Button
+        aria-label={TOOLTIP_LABEL_DEFAULT}
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/ClipboardButton.tsx:77]: The aria label here should match the tooltip.

**Review by ClaireNeveu — APPROVED** [review]
Aside from the aria label, the design looks good.

File: webui/react/src/components/kit/ClipboardButton.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  copiedMessage?: string;
+  disabled?: boolean;
+  // A get content function is used to dynamically fetch content when it is needed.
+  getContent: () => string;
+  tooltipPlacement?: Placement;
+  onCopy?: () => void;
+}
+
+export const TOOLTIP_LABEL_DEFAULT = 'Copy to clipboard';
+export const TOOLTIP_LABEL_SUCCESS = 'Copied!';
```
Comment (julian-determined-ai) [inline webui/react/src/components/kit/ClipboardButton.tsx:22]: **nit:** does this constant need to be exported?

**Review by julian-determined-ai — APPROVED** [review]
Thanks for adding the note about the copying only working on `localhost` and `https` since that has caused confusion in the past.

**Comment by hkang1** [comment]
> Thanks for adding the note about the copying only working on `localhost` and `https` since that has caused confusion in the past.

yeah for sure, I figured it was a good opportunity to really be explicit and really when I thought about it, those are the only times it would actually fail.

File: webui/react/src/components/kit/ClipboardButton.tsx (hunk 2 above)
Comment (hkang1) [reply webui/react/src/components/kit/ClipboardButton.tsx:22]: nope, I'll remove the `export`. I think I was using it for a test at one point but now it's not being used there. good catch

**Pull request merged**